modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayBuild.macro

```diff
@@ -0,0 +1,69 @@
+definition {
+
+	macro addBuild {
+		Click(locator1 = "Testray#ADD_PROJECT");
+
+		Type(
+			locator1 = "TestrayManage#NEW_OPTION_NAME",
+			value1 = "${fieldName}");
+
+		Select(
+			locator1 = "TestrayBuild#ROUTINE_ID",
+			value1 = "${routine}");
+
+		Select(
+			locator1 = "TestrayBuild#PRODUCT_VERSION",
+			value1 = "${productVersion}");
+
+		Type(
+			locator1 = "TestrayBuild#GIT_HASH_FIELD",
+			value1 = "${gitHash}");
+
+		Type(
+			locator1 = "TestrayCases#FIELD_DESCRIPTION",
+			value1 = "${Description}");
+
+		Click(locator1 = "TestrayManage#SAVE_BUTTON");
+	}
+
+	macro assertBuildDetails {
+		AssertElementPresent(
+			locator1 = "TestrayCases#CASE_DETAILS",
+			value1 = "${productVersion}");
+
+		AssertElementPresent(
+			locator1 = "TestrayCases#CASE_DETAILS",
+			value2 = "${Description}");
+
+		AssertElementPresent(
+			locator1 = "TestrayCases#CASE_DETAILS",
+			value3 = "${gitHash}");
```

Comment:
@joelrocha68 Ideally, each assert should check its own path, and not 1 generic to all asserts. Please create a locator for each element you want to assert. Example: 

CASE_DETAILS_PRODUCT_VERSION
CASE_DETAILS_DESCRIPTION
CASE _DETAILS_GIT_HASH